demo-transactions/src/main/kotlin/io/holixon/cqrshexagonaldemo/demoparent/transactions/command/application/usecase/WithdrawUseCase.kt

```diff
@@ -0,0 +1,44 @@
+package io.holixon.cqrshexagonaldemo.demoparent.transactions.command.application.usecase
+
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.application.port.inbound.account.WithdrawAccountInPort
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.application.port.outbound.account.AccountService
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.application.port.outbound.eventing.EventingOutAdapter
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.domain.model.account.Amount
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.domain.model.account.Iban
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.command.domain.model.event.MoneyWithdrawnEvent
+import io.holixon.cqrshexagonaldemo.demoparent.transactions.shared.framework.Usecase
+import mu.KLogging
+import java.math.BigDecimal
+
+@Usecase
+class WithdrawUseCase(
+    private val accountService: AccountService,
+    private val eventingOutAdapter: EventingOutAdapter
+) : WithdrawAccountInPort {
+
+    companion object : KLogging()
+
+    override fun withdraw(iban: Iban, amount: Amount) {
+        logger().info { "withdraw $amount from account ${iban.value}" }
+
+        require(amount.value >= BigDecimal.ZERO) {
+            logger().error {
+                "Amount must be positive"
+            }
+        }
+
+        val account = accountService.findAccount(iban)
+
+        val updatedAccount = accountService.withdraw(account, amount)
+
+        eventingOutAdapter.publishEvent(
+            MoneyWithdrawnEvent(
+                updatedAccount.iban,
+                updatedAccount.balance,
+                System.currentTimeMillis()
```

Comment:
This way the timestamp wouldn't be of good testablity. If a time stamp is needed by now, we should introduce a timeService, returning a `TimeStamp` domain object. This way it's mockable, we avoid the mixing of domain and non-domain objects and are of better control of the time stamp itself (e.g. If we want UTC or whatever time zone).